lib/ui/platform_dispatcher.dart

```diff
@@ -1314,6 +1371,100 @@ class WindowPadding {
   }
 }
 
+/// Area of the display that may be obstructed by a hardware feature.This is
+/// populated only on Android.
+///
+/// The [bounds] are measured in logical pixels. On devices with two screens the
+/// coordinate system starts with [0,0] in the top-left corner of the left screen
+/// and expands to include the visual space between the screens and the right screen.
+/// The [type] can be used to determine if this display feature obstructs the screen or not.
+/// For example, [DisplayFeatureType.hinge] and [DisplayFeatureType.cutout] both obstruct the display,
+/// while [DisplayFeatureType.fold] does not.
+class DisplayFeature {
+  const DisplayFeature({
+    required this.bounds,
+    required this.type,
+    required this.state,
+  });
+
+  /// Area of the view occupied by this display feature, measured in logical pixels.
+  ///
+  /// On devices with two screens, the Flutter view spans from the top-right corner
+  /// of the left screen to the bottom-right corner of the right screen, including
+  /// the area occupied by any display feature. Bounds of display features are
+  /// reported in this coordinate system.
+  ///
+  /// For example, on a dual screen device in portrait mode:
+  ///
+  /// - [bounds.left] gives you the size of left screen, in logical pixels.
+  /// - [bounds.right] gives you the size of the left screen + the hinge width.
+  final Rect bounds;
+
+  /// Type of display feature, e.g. hinge, fold, cutout.
+  final DisplayFeatureType type;
+
+  /// Posture of display feature, which is populated only for folds and hinges.
+  /// For cutouts, this is [DisplayFeatureState.unknown]
+  final DisplayFeatureState state;
+
+  @override
+  bool operator ==(Object other) {
+    return identical(this, other) || other is DisplayFeature &&
+            runtimeType == other.runtimeType &&
+            bounds == other.bounds &&
+            type == other.type &&
+            state == other.state;
+  }
+
+  @override
+  int get hashCode => hashValues(bounds, type, state);
+
+  @override
+  String toString() {
+    return 'DisplayFeature(rect: $bounds, type: $type, state: $state)';
+  }
+}
+
+/// Type of display feature (screen obstruction). This enum contains both
+/// foldable obstructions and cutout obstructions.
+///
+/// Some, like [DisplayFeatureType.fold], can be reported without actually
+/// impeding drawing on the screen. They are useful for knowing where the
+/// display is bent or has a crease. The [DisplayFeature] bounds can be
+/// 0-width in such cases.
+enum DisplayFeatureType {
```

Comment:
It would be great if you could include the pictures you have in the PR description here as well. You can upload assets to https://github.com/flutter/assets-for-api-docs and link/embed them in this PR.

Reply:
I opened PR https://github.com/flutter/assets-for-api-docs/pull/137 in order to make this happen.

Reply:
Merged! Thanks!